### PR TITLE
feat: Vector db integration

### PR DESCRIPTION
**Pull request by awtkns**

### CODE REVIEW

**Comment by byronvoorbach** [comment]
We would love to add Weaviate to this PR as well @awtkns!

**Comment by awtkns** [comment]
@byronvoorbach would you guys be able to help out with hosting 😉

**Comment by hsm207** [comment]
> @byronvoorbach would you guys be able to help out with hosting 😉

@awtkns sure. we have [WCS](https://weaviate.io/pricing).

File: src/services/agent-service.ts [hunk 1]
```diff
@@ -25,15 +26,25 @@ async function executeTaskAgent(
...
+  const ctx = await getContext({ id: "changeme", task });
+  const context = ctx.map((c) => c.document).join(", ");
+
   const completion = await new LLMChain({
     llm: createModel(modelSettings),
     prompt: executeTaskPrompt,
   }).call({
     goal,
     task,
+    context,
```
Comment (asim-shrestha) [inline src/services/agent-service.ts:38]: Do we need to pass in context for creating a task? Shouldn't this be for execute?

File: src/services/agent-service.ts [hunk 2]
```diff
@@ -25,15 +26,25 @@ async function executeTaskAgent(
   goal: string,
   task: string
 ) {
+  const ctx = await getContext({ id: "changeme", task });
```
Comment (asim-shrestha) [inline src/services/agent-service.ts:29]: Should we use the run id as the id for the context?

**Pull request closed without merging**